src/state/transactions/updater.tsx

```diff
@@ -1,25 +1,31 @@
-import { TransactionReceipt } from '@ethersproject/abstract-provider'
+import { Log } from '@ethersproject/abstract-provider'
 import { BigNumber } from '@ethersproject/bignumber'
+import { Fraction } from '@kyberswap/ks-sdk-core'
+import { ParsedTransactionMeta, ParsedTransactionWithMeta } from '@solana/web3.js'
 import { ethers } from 'ethers'
 import { findReplacementTx } from 'find-replacement-tx'
+import JSBI from 'jsbi'
 import { useCallback, useEffect, useMemo } from 'react'
 import { useDispatch, useSelector } from 'react-redux'
 
-import { AGGREGATOR_ROUTER_SWAPPED_EVENT_TOPIC } from 'constants/index'
+import { AGGREGATOR_ROUTER_SWAPPED_EVENT_TOPIC, APP_PATHS } from 'constants/index'
+import { useActiveWeb3React, useWeb3React } from 'hooks'
 import useMixpanel, { MIXPANEL_TYPE, NEED_CHECK_SUBGRAPH_TRANSACTION_TYPES } from 'hooks/useMixpanel'
-import { AppPaths } from 'pages/App'
+import { NotificationType, useBlockNumber, useTransactionNotify } from 'state/application/hooks'
 import { useSetClaimingCampaignRewardId } from 'state/campaigns/hooks'
+import connection from 'state/connection/connection'
+import { AppDispatch, AppState } from 'state/index'
+import { findTx } from 'utils'
 import { getFullDisplayBalance } from 'utils/formatBalance'
 
-import { useActiveWeb3React } from '../../hooks'
-import { NotificationType, useBlockNumber, useTransactionNotify } from '../application/hooks'
-import { AppDispatch, AppState } from '../index'
-import { SerializableTransactionReceipt, checkedTransaction, finalizeTransaction, removeTx, replaceTx } from './actions'
+import { checkedTransaction, finalizeTransaction, removeTx, replaceTx } from './actions'
+import { SerializableTransactionReceipt, TRANSACTION_TYPE, TransactionDetails } from './type'
 
-export function shouldCheck(
+function shouldCheck(
   lastBlockNumber: number,
-  tx: { addedTime: number; receipt?: SerializableTransactionReceipt; lastCheckedBlockNumber?: number },
+  tx?: { addedTime: number; receipt?: SerializableTransactionReceipt; lastCheckedBlockNumber?: number },
 ): boolean {
+  if (!tx) return false
```

Comment:
if (!tx) return false
if (tx.receipt) return false
let's combine the condition

Reply:
I prefer separate them. It's more easily to read IMO.